cmd/packaging/darwin-bundle.go

```diff
@@ -6,10 +6,15 @@ var DarwinBundleTask = &packagingTask{
 	templateFiles: map[string]string{
 		"darwin-bundle/Info.plist.tmpl": "{{.applicationName}} {{.version}}.app/Contents/Info.plist.tmpl",
 	},
-	executableFiles:               []string{},
-	buildOutputDirectory:          "{{.applicationName}} {{.version}}.app/Contents/MacOS",
-	packagingScriptTemplate:       "mkdir -p \"{{.applicationName}} {{.version}}.app/Contents/Resources\" && png2icns \"{{.applicationName}} {{.version}}.app/Contents/Resources/icon.icns\" \"{{.applicationName}} {{.version}}.app/Contents/MacOS/assets/icon.png\"",
+	executableFiles:      []string{},
+	buildOutputDirectory: "{{.applicationName}} {{.version}}.app/Contents/MacOS",
+	packagingScriptTemplate: map[string]string{
+		"linux": "mkdir -p \"{{.applicationName}} {{.version}}.app/Contents/Resources\" && png2icns \"{{.applicationName}} {{.version}}.app/Contents/Resources/icon.icns\" \"{{.applicationName}} {{.version}}.app/Contents/MacOS/assets/icon.png\"",
```

Comment:
I'm considering changing these bash scripts to functions, so that the tool lookup and execution can be defined in Go. I think it gives us better control over preparing arguments and escaping etc, and makes the code more readable. Also we can give better errors, pointing to the exact part of the packaging 'script' that failed. Thoughts?

Reply:
I think that's a good idea. Should it still be possible to detect the available tools before running?

Reply:
The go code then also wouldn't have to check if the tool is available, because it only runs if the tools are available.

Reply:
Yes, but it could be part of the packagingScriptTemplate. Which then would probably be changed to something like:
```go
type packagingDetails struct {
    applicationName string
    etc...
    workingDirectory string
}
type packagingFunction func(data *packagingDetails) error

[...]

packagingFunctions: map[string]packagingFunction {
    "linux": func(data) error { .. },
}
```

What do you think?

Reply:
I would prefer checking for the tools availability before the task runs

Reply:
Hm I'm not sure if only one function with if statements/whatever needed would be better, because some of the scripts are exactly the same on another platform.  
So I'd keep the `requiredTools` field and switch the `packagingScriptTemplate` to `packagingFunction` with a single function for every OS. Then we still need to check if an OS is supported, but for that could have a slice of supportes OSs? I'm not sure how smooth that is, but something like that would be good I think.